### ISSUE TITLE
Make a project publicly available and allow external contribution

### ISSUE THREAD

**Issue by charleygros** [biigle/core#327]
Hi there,

I've almost finish to label a dataset of images using BIIGLE, which has been a very good experience, many many thanks.

I would like to make this dataset:
- publicly available
- allow external users to contribute to it, ie add new annotations that would be, after review, integrated into the dataset

Is it something possible with BIIGLE?

Cheers,
Charley

**Comment by mzur** [comment]
Hi Charley, no, something like this is not possible right now. We thought about public projects in another context ( #53 ) before but there is no progress on that front at the moment. Allowing external contributions that need to pass review is an interesting idea. We'll discuss this with the team. Unfortunately, you can't expect this to be implemented in the near future.

**Comment by charleygros** [comment]
Hi Martin, thanks for your reply.

The common approach to make a dataset "publicly available" is to export a csv file, and drop it on Zenodo or similar.

I think we can go a step further in terms of transparency and community sharing: allow people to see how exactly your dataset was generated, and potentially correct it or add new annotations?

An alternative to "pass review" could be to "fork a project", in the same way we "fork a Github repository"?

Please keep us updated if you can :-)

**Comment by mzur** [comment]
I agree that this could be a nice feature. However, we have only limited funding and I'm currently the only active developer. If anyone on your side is willing to contribute to the BIIGLE codebase, please go ahead :wink:

You can always monitor the progress of this issue here.

**Comment by charleygros** [comment]
> If anyone on your side is willing to contribute to the BIIGLE codebase, please go ahead 😉

I will chat with the team and see what we could do! Cheers, Speak soon

**Comment by mzur** [comment]
We had a discussion about this feature with the team. Basically public projects and (public) community driven contributions would move BIIGLE more dowards a (long term) data archive and a citizen science platform. In short, we don't want BIIGLE to become a long term data archive. Of course data can be stored there as long as BIIGLE exists but not as a permanent record (e.g. with a DOI). BIIGLE should always (just) be an annotation tool. Instead, we plan to establish interfaces to external data archives whenever suitable archives are proposed.

Citizen science is a rather large topic and we are not sure yet if it can be integrated in BIIGLE in a meaningful way or if a separate tool/platform would be the way to go. This could be investigated in a dedicated research project.

What affects both ideas are concerns about copyright. Basically we (the biigle.de maintainers) could be held accountable for possible copyright infringements of published datasets. We're not sure how to handle this, yet.

Sorry to sound so negative about your ideas now but I wanted to express/document our concerns.

**Comment by charleygros** [comment]
> BIIGLE should always (just) be an annotation tool. Instead, we plan to establish interfaces to external data archives whenever suitable archives are proposed.

So, if I understand correctly, you are planning to allow BIIGLE users to (i) load a specific archived dataset (eg Zenodo) into a new BIIGLE project, possibly containing both images and annotations, (ii) and to add new annotations / modify the old annotations within this new BIIGLE project.
Is it correct? If so, I think that's a great idea and I would be interested in using this feature!

> What affects both ideas are concerns about copyright. Basically we (the biigle.de maintainers) could be held accountable for possible copyright infringements of published datasets. We're not sure how to handle this, yet.

That's perfectly understandable!

> Sorry to sound so negative about your ideas now but I wanted to express/document our concerns.

No worries at all, I understand. Thanks for your help!!

**Comment by mzur** [comment]
> So, if I understand correctly, you are planning to allow BIIGLE users to (i) load a specific archived dataset (eg Zenodo) into a new BIIGLE project, possibly containing both images and annotations

I actually meant the other way around: BIIGLE providing an export of a volume in a format that is suitable for archival. But to offer an import of the archived volume is no big deal once the export exists. We already have something like this for label trees. The images must always be handled separately/manually, though.

> (ii) and to add new annotations / modify the old annotations within this new BIIGLE project.

Yes, once you have a new volume in BIIGLE, you can do with it what you want.

This is a different idea to public projects. If you are interested in such a "public volume export", please open a new issue in [biigle/sync](https://github.com/biigle/sync).

**Comment by charleygros** [comment]
> BIIGLE providing an export of a volume in a format that is suitable for archival.

Great! That would be really valuable

> But to offer an import of the archived volume is no big deal once the export exists. We already have something like this for label trees. The images must always be handled separately/manually, though.

Good to know!!

> If you are interested in such a "public volume export", please open a new issue in [biigle/sync](https://github.com/biigle/sync).

Super, will do!!
